jaxrs/src/test/scala/com/cerner/beadledom/jaxrs/provider/ForwardedHeaderFilterSpec.scala

```diff
@@ -0,0 +1,266 @@
+      }
+
+      describe("if the protocol is not set to https"){
+        it("keeps the request http"){
+          val request = createContainerRequestContext
+          val securityContext = mock[SecurityContext]
+          Mockito.when(securityContext.isSecure).thenReturn(false)
+          Mockito.when(request.getSecurityContext).thenReturn(securityContext)
+          Mockito.when(request.getHeaderString("X-Forwarded-Proto")).thenReturn(null)
+
+          Mockito.when(request.getHeaderString("Forwarded"))
+            .thenReturn(forwardedHeaderMultiValuedString.concat(";proto=other"))
+
+          val forwardedHeaderFilter: ForwardedHeaderFilter = new ForwardedHeaderFilter
+          forwardedHeaderFilter.filter(request)
+
+          Mockito.verify(request).setRequestUri(new URI("http://hello.there"))
+        }
+      }
+
+      describe("if no protocol is set"){
+        it("keeps the request http"){
+          val request = createContainerRequestContext
+          val securityContext = mock[SecurityContext]
+          Mockito.when(securityContext.isSecure).thenReturn(false)
+          Mockito.when(request.getSecurityContext).thenReturn(securityContext)
+          Mockito.when(request.getHeaderString("X-Forwarded-Proto")).thenReturn(null)
+
+          Mockito.when(request.getHeaderString("Forwarded"))
+            .thenReturn(forwardedHeaderMultiValuedString)
+
+          val forwardedHeaderFilter: ForwardedHeaderFilter = new ForwardedHeaderFilter
+          forwardedHeaderFilter.filter(request)
+
+          Mockito.verify(request).setRequestUri(new URI("http://hello.there"))
+        }
+      }
+    }
+
+    describe("with just a protocol value"){
+      describe("if the protocol is set to https"){
+        it("changes the request to https"){
+          val request = createContainerRequestContext
+          val securityContext = mock[SecurityContext]
+          Mockito.when(securityContext.isSecure).thenReturn(false)
+          Mockito.when(request.getSecurityContext).thenReturn(securityContext)
+          Mockito.when(request.getHeaderString("X-Forwarded-Proto")).thenReturn(null)
+          Mockito.when(request.getHeaderString("Forwarded")).thenReturn("proto=https")
+
+
+          val forwardedHeaderFilter: ForwardedHeaderFilter = new ForwardedHeaderFilter
+          forwardedHeaderFilter.filter(request)
+
+          Mockito.verify(request).setRequestUri(new URI("https://hello.there"))
+        }
+      }
+
+      describe("if the protocol is not set to https"){
+        it("keeps the request http"){
+          val request = createContainerRequestContext
+          val securityContext = mock[SecurityContext]
+          Mockito.when(securityContext.isSecure).thenReturn(false)
+          Mockito.when(request.getSecurityContext).thenReturn(securityContext)
+          Mockito.when(request.getHeaderString("X-Forwarded-Proto")).thenReturn(null)
+          Mockito.when(request.getHeaderString("Forwarded")).thenReturn("proto=other")
+
+
```

Comment:
nit: extra line can be removed

Reply:
removed: https://github.com/cerner/beadledom/pull/149/commits/ccaf18887d21e0d9f5f736d0dd152fb8e436ecdf